website/addons/wiki/templates/edit.mako

```diff
@@ -111,7 +111,7 @@
               </div>
           </div>
 
-          % if user['can_edit']:
+          % if has_permission_or_publicly_editable:
```

Comment:
Indeed, the `wiki_settings` object won't be available in all the places `_view_project` is called.

I wasn't suggesting that you add this variable to `_view_project`; just add it to the wiki view that uses this template

```python
ret.update(_view_project(...))
ret['user']['can_edit_wiki'] = can_edit_body
```